source/feature/openvswitch/openvswitch-spec.html.md

```diff
@@ -0,0 +1,183 @@
+
+### Persistence
+OVS persistency is embedded in the switch.
+IP settings are not included and may be implemented using the current persistence scheme.
+- Note: An alternative may be to use a minimal persistent management settings and zero persistency for all the rest. Nevertheless these changes are out of scope and are to be handled under a zero persistency feature. 
+
+### Rollback
+VDSM networking uses rollback to recover from a setup request which faild to complete.
+The network API setupNetworks verb has a complex rollback semantics.
+In the vdsm api context, a rollback is detected and a command is issued to supervdsm for handling. This function is currently unused, because the ifcfg configurator never triggers it.
+Handling the rollback as close as possible to the setupNetworks api implementation seems more reasonable, moving this rollback handler to the network api (currently in supervdsm context) is to be considered.
+The OVS rollback is to be integrated into the existing rollback functionality without heavely affecting existing beahaviours:
+
+- OVS configurators should trigger a rollback which is to be handled at the api level (as described above).
+- The ifcfg configurator should keep the existing logic, but in addition it should inform the api level of the rollback which it handled.
+Moving the handling of the rollback from the ifcfg configurator to the api level should be investigated, simplifying rollback logic.
+- The api level should combine both ifcfg and ovs results in terms of the rollback, and issue relevant commands. TBD
+
+### VDSM API
+Minimal changes in current API verbs are required:
+
+- setupNetworks: Per network, add a net-switch type.
+  - Currently it can have two values: ‘legacy’ and ‘ovs’.
+  - If not explicitly set, assume ‘legacy’. (for compatability with Endgine 3.6)
+- oVirt currently differentiates between VM (bridged) networks and non-VM networks due to historical performance benefit of non-bridged networks for non-VM payloads (such as storage and migration). This differentiation is moot with OVS, but in order to keep Engine oblivious to the change, Vdsm should lie to it, and report bridged=False for networks that requested it. Dropping the differentiation from Engine should happen in a later stage.
+
+### Upgrade from native to OVS networking
+Transition between networking implementation must be supported for the host as a whole. Nevertheless, if the user has defined his own networking configuration which is not controlled by VDSM, it should coexist with VDSM controlled networks.
+
+Mixing between native and OVS controlled networks is not initially supported, therefore, when a network is marked with a different type from the existing, a validation check should make sure that all other networks are marked for the same network type.
+
+The following sequence describes the transition steps @Engine:
+
+- Hosts are upgraded to OVS supported version (4.0).
```

Comment:
"""
Vdsm on hosts is upgraded to an OVS-supporting version (4.0). No change takes place to preexisting network configuration, which would still be bridge-based.
"""